corehq/apps/app_manager/util.py

```diff
@@ -359,3 +380,31 @@ def get_commcare_versions(request_user):
     versions = [i.build.version for i in CommCareBuildConfig.fetch().menu
                 if request_user.is_superuser or not i.superuser_only]
     return sorted(versions, key=version_key)
+
+
+def get_usercase_keys(items):
+    n = len(USERCASE_PREFIX)
+    return {k[n:]: v for k, v in items if k.startswith(USERCASE_PREFIX)}
+
+
+def get_usercase_values(items):
+    n = len(USERCASE_PREFIX)
+    return {k: v[n:] for k, v in items if v.startswith(USERCASE_PREFIX)}
+
+
+def any_usercase_items(items):
+    return any(i.startswith(USERCASE_PREFIX) for i in items)
+
+
+def actions_use_usercase(actions):
+    return (any_usercase_items(actions['update_case']['update'].iterkeys()) or
+            any_usercase_items(actions['case_preload']['preload'].itervalues()))
+
+
+def enable_usercase(domain_name):
+    with CriticalSection([domain_name + '_enable_usercase']):
```

Comment:
No need to block on this, but I've been using the convention of 'keybase_%s' % identifier for the key names (i.e., key base first, then unique object identifier) to make it harder to have key name conflicts.